### DISCUSSION TITLE
QMC5883L spontaneously stops responding

### DISCUSSION THREAD

**Discussion by jac2424** [tronikos/esphome-magnetometer-water-gas-meter#25] [General]
Running into an issue with my gas meter where  the QMC5883L seems to randomly stop sending data every few days. If I reinstall the flash ESP32 logs show QMC5883L as failed status. If I disconnect the power to the ESP32 and connect again after few seconds, it starts working again as normal. Not sure if the issue is with the QMC5883L or the ESP32. Any ideas how I can look into the cause as I have the same setup with the water meter and that one does not have this issue?

**Comment by tronikos** [comment]
How long is the wire? Is there anything along the wire that could interfere with the signal? I'd use CAT6 and 5V since it's more reliable. I'd recheck the wiring and maybe resolder them. If it still fails I'd just replace both the ESP chip and the magnetometer. Luckily they are both very cheap.

**Reply by jac2424** [reply]
The wire is about 20 feet but instead of CAT6 cable I used thermostat wire as I figured it would stand up better to the weather. I didn't solder them in as I wanted to get it stable first so I used Dupont Wires on both ends and the thermostat wire in between. Same exact wiring seems to work fine for the water meter with no issues but for gas meter I verified wires were still intact but QMC5883L was failed status. With no tweaks to wiring, just a power reset on ESP32, it starts working again. I do have a spare ESP32 so will try to swap it next time it happens as you suggested but it seems I have to remove and readd the device in home assistant for that since the MAC ID changes and home assistant does not pick it up even with same config and naming. If that does not work, will solder it in.

**Reply by tronikos** [reply]
That explains it. For i2c communication, longer wires can introduce noise and signal loss, potentially causing communication errors or instability. You really need to be using twisted pair cables with proper shielding and lower capacitance such as CAT6.

**Reply by jac2424** [reply]
Swapped out the thermostat wire for CAT6, was a paint to solder as pin headers were already on ESP32 and QMC5883L boards. Will report back if issue still occurs with the new wiring.

**Comment by 42Network** [comment]
Hello. I have this exact problem where I get data for a few days fine, but it stops and I see the QMC5883L failed status. Although in my case I am using just 10 feet of ground burial-rated CAT6. I did group all the unused wires together for ground, as suggested in the build.

Pressing the reset button on the ESP32 module gets it working again for me, but just for a few days. I was thinking of scheduling a daily restart of the ESP32, but this is obviously a kludge. 

I thought I would catch it by looking for the sensors going unavailable but they don't -  they just show last updated e.g. three days ago.

Do you have any other suggestions?

**Comment by jac2424** [comment]
Mine started getting unstable again with CAT 6 wire as well (did seems to last longer between becoming unavailable) but I suspected poor solder connection on ESP32. I swapped out ESP32 and now have it connected with below board to see if it gets stable.

https://www.amazon.ca/gp/aw/d/B0CCVSJYJG?psc=1&ref=ppx_pop_mob_b_asin_title

**Comment by StarScream159** [comment]
I too am having this issue with it working and then stopping. Mine will work for a few hours only and then stop. I've got it connected to 5V and whites connected to ground as instructed. I do have a long wire though, [50m but it is CAT6](https://www.amazon.ca/Ethernet-Higher-Bandwidth-Internet-Network/dp/B017P34WZI). It is only 30 AWG though not sure if that would effect it. It's odd because I figured it wouldn't work at all (due to the length) or it would just work. Not both. Is the wire length really my problem here?

**Reply by jac2424** [reply]
Mine has been stable since I swapped out the ESP32 board and used the ESP32 Breakout Board to connect the wires and ESP32 just over a week ago. Will see if it remains stable. My cable run is about only 20ft.
